### PR TITLE
Cor 33565

### PR DESCRIPTION
**Pull request by barakfireblocks**
added getPaginatedAddresses 
// Get addresses for a vault account for asset with paging

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1155,6 +1155,33 @@ export interface MaxBip44IndexUsedResponse {
...
+
+export interface PaginatedAddressesResponse {
+    addresses: AddressResponse[];
+    paging?: {
+        before?: string;
+        after?: string;
+    };
+}
+
+export interface OptionalPaginatedAddresseRequestFilters {
```
Comment (ranmitel) [inline src/types.ts:1180]: -> OptionalPaginatedAddressesRequestFilters
(was missing 's' on Addresses)

File: src/fireblocks-sdk.ts [hunk 2]
```diff
@@ -1159,6 +1161,13 @@ export class FireblocksSDK {
         return await this.apiClient.issueGetRequest(url);
     }
 
+    /**
+     * Get addresses for a vault account for asset with paging
+     */
+        public async getPaginatedAddresses(accountId: string, assetId: string, paginatedAddresseRequestFilters?: OptionalPaginatedAddresseRequestFilters): Promise<PaginatedAddressesResponse> {
```
Comment (ranmitel) [inline src/fireblocks-sdk.ts:1167]: -> paginatedAddressesRequestFilters
(missing 's' for Addresses)

Comment (ranmitel) [inline src/fireblocks-sdk.ts:1167]: I believe you named this arg in our yaml spec (Doc) as vaultAccountId, so better align on this here

File: src/fireblocks-sdk.ts [hunk 3]
```diff
@@ -22,6 +22,8 @@ import {
     InternalWalletAsset,
     MaxSpendableAmountResponse,
     MaxBip44IndexUsedResponse,
+    PaginatedAddressesResponse,
```
Comment (ranmitel) [inline src/fireblocks-sdk.ts:25]: what about AddressResponse ?

File: src/fireblocks-sdk.ts [hunk 4]
```diff
@@ -1159,6 +1161,13 @@ export class FireblocksSDK {
         return await this.apiClient.issueGetRequest(url);
     }
 
+    /**
+     * Get addresses for a vault account for asset with paging
+     */
+        public async getPaginatedAddresses(accountId: string, assetId: string, paginatedAddresseRequestFilters?: OptionalPaginatedAddresseRequestFilters): Promise<PaginatedAddressesResponse> {
+            return await this.apiClient.issueGetRequest(`/v1/vault/accounts/${accountId}/${assetId}/addresses_paginated?${queryString.stringify(paginatedAddresseRequestFilters)}`);
```
Comment (ranmitel) [inline src/fireblocks-sdk.ts:1168]: queryString.stringify(...) works ok when paginatedAddresseRequestFilters is undefined ?

File: src/fireblocks-sdk.ts (hunk 3 above)
Comment (barakfireblocks) [reply src/fireblocks-sdk.ts:25]: i can import it but there are not use in it,
i only use PaginatedAddressesResponse as the return value of the function:
Promise<PaginatedAddressesResponse>
and that contain the AddressResponse as an array

File: src/fireblocks-sdk.ts (hunk 4 above)
Comment (barakfireblocks) [reply src/fireblocks-sdk.ts:1168]: yes, test it with only vaultAccountId and assetId and it works

File: src/fireblocks-sdk.ts (hunk 3 above)
Comment (ranmitel) [reply src/fireblocks-sdk.ts:25]: Ok, if that's not a must -leave it as is.

File: src/fireblocks-sdk.ts (hunk 4 above)
Comment (ranmitel) [reply src/fireblocks-sdk.ts]: great

File: src/fireblocks-sdk.ts [hunk 5]
```diff
@@ -1159,6 +1161,13 @@ export class FireblocksSDK {
         return await this.apiClient.issueGetRequest(url);
     }
 
+    /**
+     * Get addresses for a vault account for asset with paging
```
Comment (ranmitel) [inline src/fireblocks-sdk.ts:1165]: let's use something similar to what u used in dev-api \ yaml (for the Doc) : 

"Gets a paginated response of the addresses for a given vault account and asset"

**Review by yarinvak — CHANGES_REQUESTED** [review]
change MR title to be descriptive and without the jira ticket ref

**Pull request closed without merging**